web/src/js/script.js

```diff
@@ -0,0 +1,395 @@
+                    $(value).prop("checked", true);
+                });
+            }
+        };
+
+        $.ajax({
+                   url: "/protocols",
+                   type: "GET",
+                   contentType: 'application/json',
+                   success: fillProtocolData
+               });
+    };
+
+    var inputGet = function (item_id) {
+        return $('.config-item[data-field="' + item_id + '"]').val();
+    };
+
+    var inputGetInt = function (item_id) {
+        return parseInt(inputGet(item_id));
+    };
+
+    var checkboxGet = function (item_id) {
+        return $('.config-item[data-field="' + item_id + '"]').prop("checked");
+    };
+
+    var protocolGet = function (item_id) {
+        var checked = $('.protocols-item:checked');
+        if ($('.protocols-item').length === checked.length) {
+            return [];
+        }
+        return $.map(checked, function (x) {
+            return $(x).val();
+        });
+    };
+
+    function ConfigItem(name, factory, apply, fetch, help) {
+        this.name = name;
+        this.factory = factory;
+        this.apply = apply;
+        this.fetch = fetch;
+        this.help = help;
+    }
+
+    function GroupItem(name, factory) {
+        this.name = name;
+        this.factory = factory;
+    }
+
+    var CONFIG_ITEMS = [
+        new GroupItem("General Config", legendFactory),
+        new ConfigItem("deviceName", inputFieldFactory, inputApply, inputGet, "The general name of the device"),
+        new ConfigItem("mdnsName", inputFieldFactory, inputApply, inputGet, "The name the device shall announce via mdns"),
+        new ConfigItem("configPassword", passwordFieldFactory, inputApply, inputGet, "The admin password for the web UI"),
+
+        new GroupItem("MQTT Config", legendFactory),
+        new ConfigItem("mqttBroker", inputFieldFactory, inputApply, inputGet, "MQTT Broker host"),
+        new ConfigItem("mqttBrokerPort", inputFieldFactory, inputApply, inputGetInt, "MQTT Broker port"),
+        new ConfigItem("mqttUser", inputFieldFactory, inputApply, inputGet, "MQTT username (optional)"),
+        new ConfigItem("mqttPassword", passwordFieldFactory, inputApply, inputGet, "MQTT password (optional)"),
+        new ConfigItem("mqttRetain", checkboxFactory, checkboxApply, checkboxGet, "Retain MQTT messages"),
+
+        new GroupItem("MQTT Topic Config", legendFactory),
+        new ConfigItem("mqttReceiveTopic", inputFieldFactory, inputApply, inputGet, "Topic to publish received signal"),
+        new ConfigItem("mqttSendTopic", inputFieldFactory, inputApply, inputGet, "Topic to get signals to send from"),
+
+        new GroupItem("433MHz RF Config", legendFactory),
+        new ConfigItem("rfEchoMessages", checkboxFactory, checkboxApply, checkboxGet, "Echo sent rf messages back"),
+        new ConfigItem("rfReceiverPin", inputFieldFactory, inputApply, inputGetInt, "The pin used for the rf receiver"),
```

Comment:
Please make clear, this are the GPIO pins.

Reply:
done in the next push